### PR TITLE
WIP: Extend Baseline agent-installer steps for the BM labs automation - Add disconnected configuration

### PR DESCRIPTION
**Pull request by bmanzari**
Add support for restricted network environment configurations in agent based installations on bare metal

### CODE REVIEW

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 1]
```diff
@@ -169,9 +173,31 @@ API_VIP="$(yq ".api_vip" "${SHARED_DIR}/vips.yaml")"
 INGRESS_VIP="$(yq ".ingress_vip" "${SHARED_DIR}/vips.yaml")"
 mkdir -p "${INSTALL_DIR}"
 
-echo "Installing from initial release ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
-oc adm release extract -a "$PULL_SECRET_PATH" "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}" \
+if [ "${DISCONNECTED}" == "true" ]; then
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:176]: The command can continue to be extracted as before.

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh [hunk 2]
```diff
@@ -15,6 +15,10 @@ trap 'echo "$?" > "${SHARED_DIR}/install-status.txt"' TERM ERR
 [ -z "${workers}" ] && { echo "\$workers is not filled. Failing."; exit 1; }
 [ -z "${masters}" ] && { echo "\$masters is not filled. Failing."; exit 1; }
 
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
```
[the remark this answers is not in the document]
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh:18]: added to upi and ipi

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh (hunk 1 above)
Comment (bmanzari) [reply ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh]: dropped if, the command will extract as before using override

**Comment by aleskandro** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

File: ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-ref.yaml [hunk 3]
```diff
@@ -33,6 +33,14 @@ ref:
   - name: AUX_HOST
     default: ""
     documentation: ""
+  - name: DISCONNECTED
+    default: "false"
+  - name: OCP_REPO
```
Comment (bmanzari) [inline ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-ref.yaml:38]: TODO: remove

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

File: ci-operator/step-registry/baremetal/lab/pre/firewall/baremetal-lab-pre-firewall-commands.sh [hunk 4]
```diff
@@ -30,12 +30,32 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
   IP_ARRAY+=( "$ip" )
 done
 
+echo "${IP_ARRAY[@]}"
+
 timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s -- \
-  "${IP_ARRAY[@]}" << 'EOF'
+  "${INTERNAL_NET_CIDR}" "${DOCKER_NET_CIDR}" "${IP_ARRAY[@]}"  << 'EOF'
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/pre/firewall/baremetal-lab-pre-firewall-commands.sh:36]: Unbound variable?

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-compact-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-dualstack-dynamic-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-dualstack-dynamic-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-dualstack-dynamic-connected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14

**Comment by bmanzari** [comment]
100% Tests Passed build https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37221/rehearse-37221-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-agent-baremetal-sno-ipv4-static-disconnected-p1-f14/1638929067065479168

**Review by aleskandro — COMMENTED** [review]
Added a few comments. Also, you need to do some cleaning up of your branch (53 commits currently):
1. drop unneeded commits
2. squash the others by "logical groups"
3. set proper commit messages
4. rebase from origin/master

File: ci-operator/step-registry/baremetal/lab/nfs-provisioner/baremetal-lab-nfs-provisioner-ref.yaml [hunk 5]
```diff
@@ -18,6 +18,12 @@ ref:
         The NFS_SERVER that exports a share to be used by the node. It's set by default to the same value as the AUX_HOST. 
         The NFS_SERVER must exports the /opt/nfs/${CLUSTER_NAME} folder, to be consumed by the provisioner, and 
         assumed as already existing.
+    - name: NFS_CLIENT_MIRRORED_IMAGE_URL
+      default: "quay.io/openshifttest/nfs-subdir-external-provisioner@sha256:3036bf6b741cdee4caf8fc30bccd049afdf662e08a52f2e6ae47b75ef52a40ac"
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/nfs-provisioner/baremetal-lab-nfs-provisioner-ref.yaml:22]: No need to add it here as we don't expect (and we don't want actually) users to change the value from the test configs. A variable in the script and a comment above it is enough.

File: ci-operator/step-registry/baremetal/lab/pre/firewall/baremetal-lab-pre-firewall-commands.sh [hunk 6]
```diff
@@ -30,12 +30,31 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
   IP_ARRAY+=( "$ip" )
 done
 
+echo "${IP_ARRAY[@]}"
+
 timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s -- \
-  "${IP_ARRAY[@]}" << 'EOF'
+  "${INTERNAL_NET_CIDR}" "${IP_ARRAY[@]}"  << 'EOF'
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/pre/firewall/baremetal-lab-pre-firewall-commands.sh:36]: additional space not needed before the `<<`

File: ci-operator/step-registry/baremetal/lab/post/firewall/baremetal-lab-post-firewall-commands.sh [hunk 7]
```diff
@@ -30,10 +30,12 @@ done
 
 echo 'Deprovisioning firewall configuration'
 timeout -s 9 10m ssh "${SSHOPTS[@]}" "root@${AUX_HOST}" bash -s -- \
-  "${IP_ARRAY[@]}" << 'EOF'
+  "${INTERNAL_NET_CIDR}" "${IP_ARRAY[@]}"  << 'EOF'
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/post/firewall/baremetal-lab-post-firewall-commands.sh:33]: additional space not needed before the `<<`

File: ci-operator/step-registry/baremetal/lab/nfs-provisioner/baremetal-lab-nfs-provisioner-commands.sh [hunk 8]
```diff
@@ -104,6 +107,8 @@ more ${DIR}/*.yaml | cat
 echo
 oc apply -k ${DIR}
 echo "Waiting up to 10 minutes for the nfs-provisioner pod to become ready..."
+
```
Comment (aleskandro) [inline ci-operator/step-registry/baremetal/lab/nfs-provisioner/baremetal-lab-nfs-provisioner-commands.sh:110]: additional empty rows not needed.

**Pull request closed without merging**